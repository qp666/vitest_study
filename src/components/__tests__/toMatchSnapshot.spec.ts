import { describe, it, expect } from 'vitest'
/* toMatchSnapshot 快照断言会生成一个文件，它适合一些大型的，长久不变更的地方，
例如对配置文件进行快照或者如果使用一些远程图标库 icon，我们可以对 icon 地址进行快照，
这个文件如果变更了，就会出现报错，报错就代表有风险，需要谨慎操作。 */

it('test toMatchSnapshot', () => {
  const config = { url: 'url', domain: 'domain', analysis: 'analysis alias' }
  expect(config).toMatchSnapshot()
})
