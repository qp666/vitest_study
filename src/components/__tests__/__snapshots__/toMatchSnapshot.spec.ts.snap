// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test toMatchSnapshot 1`] = `
{
  "analysis": "analysis alias",
  "domain": "domain",
  "url": "url",
}
`;
